extensions/iceberg/src/main/java/io/deephaven/iceberg/util/IcebergCatalogAdapter.java

```diff
@@ -119,21 +120,141 @@ static io.deephaven.qst.type.Type<?> convertPrimitiveType(@NotNull final Type ic
             case FIXED: // Fall through
             case BINARY:
                 return io.deephaven.qst.type.Type.find(byte[].class);
+            case UUID: // Fall through
+            case STRUCT: // Fall through
+            case LIST: // Fall through
+            case MAP: // Fall through
             default:
                 throw new TableDataException("Unsupported iceberg column type " + typeId.name());
         }
     }
 
-    @SuppressWarnings("unused")
+    public List<Namespace> listNamespaces() {
+        return listNamespaces(Namespace.empty());
+    }
+
+    public List<Namespace> listNamespaces(@NotNull Namespace namespace) {
+        if (catalog instanceof org.apache.iceberg.catalog.SupportsNamespaces) {
+            final SupportsNamespaces nsCatalog = (SupportsNamespaces) catalog;
+            return nsCatalog.listNamespaces(namespace);
+        }
+        throw new UnsupportedOperationException(String.format(
+                "%s does not implement org.apache.iceberg.catalog.SupportsNamespaces", catalog.getClass().getName()));
+    }
+
+    public Table listNamespacesAsTable() {
+        return listNamespacesAsTable(Namespace.empty());
+    }
+
+    public Table listNamespacesAsTable(@NotNull Namespace namespace) {
+        final List<Namespace> namespaces = listNamespaces();
+        final long size = namespaces.size();
+
+        // Create and return a table containing the namespaces as strings
+        final Map<String, ColumnSource<?>> columnSourceMap = new LinkedHashMap<>();
+
+        // Create the column source(s)
+        final WritableColumnSource<String> namespaceColumn =
+                ArrayBackedColumnSource.getMemoryColumnSource(size, String.class, null);
+        columnSourceMap.put("namespace", namespaceColumn);
+
+        final WritableColumnSource<Object> objectColumn =
+                ArrayBackedColumnSource.getMemoryColumnSource(size, Object.class, Namespace.class);
+        columnSourceMap.put("namespace_object", objectColumn);
+
+        // Populate the column source(s)
+        for (int i = 0; i < size; i++) {
```

Comment:
Pre-filling arrays is probably more efficient, if you're going to go the immutable source route.